File: abstractions/java/lib/src/main/java/com/microsoft/kiota/serialization/ParsableFactory.java
package com.microsoft.kiota.serialization;

import javax.annotation.Nonnull;

/**
 * Defines the factory for creating parsable objects.
 * </summary>
 * @param T The type of the parsable object.
 */
@FunctionalInterface
public interface ParsableFactory<T extends Parsable> {
    /**
     * Create a new parsable object from the given serialized data.
     * @param parseNode The node to parse use to get the discriminator value from the payload.
     * @returns The parsable object.
     */
    T Create(@Nonnull final ParseNode parseNode);
}

File: abstractions/php/src/Serialization/ParsableFactory.php
<?php

namespace Microsoft\Kiota\Abstractions\Serialization;

interface ParsableFactory {
    /**
     * Create a new parsable object from the given serialized data.
     * @param ParseNode $parseNode The node to parse use to get the discriminator value from the payload.
     * @returns Parsable The parsable object.
     */
    public function create(ParseNode $parseNode): Parsable;
}